### DISCUSSION TITLE
Superchlorinating not working

### DISCUSSION THREAD

**Discussion by mzuniga51** [tagyoureit/nodejs-poolController#433] [General]
@rstrouse Hey, remember you solved my superchlorinating command not working?  At some point it stopped working again.  This is what is happening.  I press twice on the auxiliary chlorinator and twice on the main.  Thanks.
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/8614019/replay.zip)

**Comment by rstrouse** [comment]
Make sure you have the latest software for both njsPC and dashPanel.  I just tried it on my IntelliCenter and it worked fine.

**Comment by rstrouse** [comment]
Did you delete the Aux RS485 port that was set up for the second chlorinator.  Two chlorinators cannot exist on the same RS485 bus and must be assigned to different ports.  If don't have this set up this way it will make IntelliCenter all confused since there is no addressing on IntelliChlor.  There is only one port defined in your system and the second chlorinator is set up to be read only.

Did you restore a prior backup or something?

**Comment by mzuniga51** [comment]
Will do.  Thanks!

**Comment by mzuniga51** [comment]
No restore.  I do have each one on a different port

**Reply by rstrouse** [reply]
There was no second port in your replay files and the njsPC Aux Controller interface was enabled.  You do not need that second instance of njsPC.  Under configuration you can add a second comm port and communicate through that.

To create an Aux Port
1. Install the hardware for the second port
2. Connect the chlorinator to this second port
3. Add an Aux RS485 port to the configuration
![image](https://user-images.githubusercontent.com/47839015/166612485-a5140f4d-aa96-4285-8a96-7d0b77a30497.png)


To set up the chlorinator to use the aux Port.
1. Delete the current chlorinator that is accessed by the External njsPC
2. Add a new chlorinator but make sure you select **Nixie** from the dropdown.
![image](https://user-images.githubusercontent.com/47839015/166612757-d4a5b436-8d8d-4dc2-a350-282af8df45e7.png)
4. Select the Aux 1 in the Port dropdown and press save.
![image](https://user-images.githubusercontent.com/47839015/166612921-61eef4e5-0351-4e1b-8d38-a8413a1f9e06.png)


There should be no need or desire for this second instance of njsPC.  All of it including the second chlorinator should be operated from the same njsPC instance.  Don't forget to set one for the pool body and one for the spa body.  My screenshot didn't do that.

**Comment by mzuniga51** [comment]
Thank you @rstrouse.  It works like a charm with the new auxiliary.  I deleted the other instance of njsPC.   Now, the aux cell (which in my case is Pool) is accepting the superchlor command but the main (Spa in my case) is not.   I am fully updated and caches are clear.  Here is what I am seeing
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/8627830/replay.zip)

**Comment by rstrouse** [comment]
Very odd that the njsPC implementation of superchlor is working but IntelliCenter is not.  The first thing I noticed is that the body setting for the IntelliChlor attached to IntelliCenter is tagged to the first body.  This is incorrect and will cause IntelliCenter to be confused about which body should be engaged for the chlorinator.

These must assigned correctly.  So the Nixie controlled chlorinator must be assigned to the pool and the IntelliCenter controlled chlorinator must be assigned to the spa.  I see njsPC issuing the command to IntelliCenter and it responds OK.  Then just blows it off.  This is similar to what happens when it detects that the body is not currently running.

Now if setting the bodies does not work then there is an off chance that the chlorinator in an i10D only works with the first body.  I know you have this set up backwards so you may need IntelliCenter to control the pool iChlor and let Nixie take on the spa iChlor.   They really screwed the pooch when they didn't create addressing for the IntelliChlor here.

**Comment by mzuniga51** [comment]
Yes.  Both chlorinators were set to Pool.  Given my backwards
configuration, if I set the main chlorinator to Spa it will not work (given
that the Spa circuit is not on) and if I set the aux chlorinator to Spa it
obviously don't get any reading at njsPC.   So given my backwards
configuration works so well to be able to have egg timers on the spa
circuit I guess I will have to leave without the option to superchlorinate
in the main chlorinator.  A shame since the spa is where I most frequently
need the superchlor but i guess you can't have it all.  Thanks a
bunch anyway!

**Reply by rstrouse** [reply]
All you have to do is go into the configuration and set the spa chlorinator to spa and the pool chlorinator to pool.  Yes IntelliCenter will not allow you to superchlorinate unless the particular body is on.  If it isn't it assumes there is no flow.  So turn on the spa and hit superchlorinate and it should work.

**Reply by rstrouse** [reply]
If it doesn't then switching the bodies around will do the trick.  The reason is because njsPC has no such limitation to the first body.

**Comment by mzuniga51** [comment]
Yes, but that is already how it is.  Remember you helped me set the system
so that when Spa is on it will superchlorinate.   But whenever Spa is on
heater is also on so that will not quite work for me unfortunately

**Comment by rstrouse** [comment]
You know that is a setting in IntelliCenter right?  Unfortunately, you cannot trigger the chlorinator in IntelliCenter without the body being engaged.  And since you have declared your primary body as Spa IntelliCenter is in control of that and cannot be overridden.  If you set Spa Manual Heat to on then it will not turn on the heater unless you ask it to.  I think it deselects the heater when the body turns off.

**Comment by mzuniga51** [comment]
Right, that is a good idea.  I'll try and see if it works. Thanks!

**Comment by mzuniga51** [comment]
Hey, long time no see.  I'm having an issue after I updated to 8.0.0.   If
you remember i have two cells, the one that goes to the pull connected to
the Nixie Aux1 port which is working fine.  The other is the Spa's and is
connected to the Pool port on the main instance.   I used to be able to
control that last cell from njsPC.  Now I can only control it from the
Intellicenter panel.  Every change I made in chlorination % at the panel i
can see it in njsPC but if I move the percentages in njsPC the change is
not made on the cell,

I have no idea what I'm doing wrong or if it is just a bug on the last
version.

Thanks and regards,

Manuel

**Comment by rstrouse** [comment]
Please include a replay so I can look at the comms.

**Comment by mzuniga51** [comment]
Sure.  So here I'm first trying to increase the chlorine output of the Spa
via njsPC.  Nothing happens at the cell level.  Next you will see me
changing the chlorine output at the panel.  Not only the cell then obeys
but also the change is immediately registered in njsPC.

**Reply by tagyoureit** [reply]
Unfortunately, email responses don't include the file attachments (or images).  You'll need to attach them through the browser.

**Reply by mzuniga51** [reply]
sorry. here.
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/10242029/replay.zip)

**Comment by rstrouse** [comment]
Did you delete and re-add the chlorinators at some point.  You have one chlorinator (the one that is working), controlled by Nixie and the other is set up as being controlled externally.  Not by Nixie or IntelliCenter.  To fix this delete the Spa IChlor leave the Pool IChlor alone.  Then reload your configuration.  It already exist in IntelliCenter but you told njsPC that it was controlled by other means.

**Comment by mzuniga51** [comment]
Got it.  What do you mean by "reload the configuration" exactly? Can you
please remind me how do I do that?  Thanks

**Reply by rstrouse** [reply]
First do the delete the Spa IChlor in dashPanel then go into IntelliCenter and save it to see if it re-adds it to njsPC.

**Comment by mzuniga51** [comment]
Ok. I'll try it tomorrow and let you know.  Thanks!

**Comment by mzuniga51** [comment]
Yep.  That did the trick.  Thanks!
